subsys/testsuite/include/negative_handler.h

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2020 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <zephyr.h>
+
+
+#if defined(CONFIG_NEGATIVE_TEST_FATAL_HANDLER)
+/* These valid flag used to figure out if the fault was expected or not. */
+extern volatile bool valid_fault;
+extern void *fatal_hook_param;
+
+/**
+ * @brief Set the flag that treat fatal error happened as expected
+ *
+ * This is used for negative test cases which triggers a fatal error.
+ * Set the param true will still pass the test case when expected fatal
+ * error happened. For normal test case, set it false makes it back to
+ * normal behavior.
+ *
+ * @param valid flag indicate fault is expected
+ */
+static inline void set_fault_valid(bool valid)
+{
+	valid_fault = valid;
+	/* Put a barrier here, such that no instructions get ordered by the
+	 * compiler before we set valid_fault. This can happen with expansion
+	 * of inline syscall invocation functions.
+	 */
+	compiler_barrier();
```

Comment:
Same comment, no compiler_barrier() needed for valid_fault, it's already volatile.

Reply:
Got it, thanks! will get them fixed.